zerver/tests/test_custom_profile_data.py

```diff
@@ -576,6 +576,17 @@ def test_update(self) -> None:
         )
         self.assert_json_error(result, "field_data is not a dict")
 
+        field_data = orjson.dumps(
+            {
+                "1": {"order": "2", "text": "Not Emacs"},
+            }
+        ).decode()
+        result = self.client_patch(
+            f"/json/realm/profile_fields/{field.id}",
+            info={"field_data": field_data},
+        )
+        self.assert_json_error(result, "The value of an existing select option cannot be changed.")
+
```

Comment:
I'm confused as to what criteria we use to decide whether a test lives inside `test_update` and whether it should be part of the new function. I've kept it as part of the `test_update` for now.

Reply:
Yes, that sounds good.

Reply:
I've added `test_update_field_data` and `test_update_display_in_profile_summary` to make the `test_update` function shorter. In order to be consistent with the pattern in `test_create`, I have not added different functions for `name`, `hint`, etc, since they have similar error and success states to each other. I also added a cleanup commit at the end for a code block I think was redundant